implementations/micrometer-registry-otlp/src/main/java/io/micrometer/registry/otlp/OtlpMeterRegistry.java

```diff
@@ -497,6 +503,76 @@ static double[] getSloWithPositiveInf(DistributionStatisticConfig distributionSt
         return sloWithPositiveInf;
     }
 
+    private static HistogramFlavor histogramFlavorPerMeter(OtlpConfig config, Meter.Id id) {
+        for (Map.Entry<String, HistogramFlavor> entry : config.histogramFlavorPerMeter().entrySet()) {
+            if (id.getName().startsWith(entry.getKey())) {
+                return entry.getValue();
+            }
+        }
+        return config.histogramFlavor();
+    }
+
+    private static Integer maxBucketsPerMeter(OtlpConfig config, Meter.Id id) {
+        for (Map.Entry<String, Integer> entry : config.maxBucketsPerMeter().entrySet()) {
+            if (id.getName().startsWith(entry.getKey())) {
+                return entry.getValue();
+            }
+        }
+        return config.maxBucketCount();
+    }
+
+    /**
+     * Overridable lookup mechanism for {@link HistogramFlavor}.
+     *
+     * @since 1.15.0
+     */
+    @FunctionalInterface
+    public interface HistogramFlavorPerMeterLookup {
+
+        /**
+         * Looks up the histogram flavor to use on a per-meter level. This will override
+         * the {@link OtlpConfig#histogramFlavor()} for matching Meters. The default
+         * implementation in {@link OtlpMeterRegistry} does a prefix match on the Meter's
+         * name in the map that {@link OtlpConfig#histogramFlavorPerMeter()} returns. This
+         * means that {@link OtlpConfig#histogramFlavorPerMeter()} provides the data while
+         * this method provides the logic for the lookup, and you can override them
+         * independently.
+         * @param id the {@link Meter.Id} the {@link HistogramFlavor} is configured for
+         * @return the histogram flavor mapped to the {@link Meter.Id}
+         * @see OtlpConfig#histogramFlavorPerMeter()
+         * @see OtlpConfig#histogramFlavor()
+         */
+        HistogramFlavor getHistogramFlavor(OtlpConfig config, Meter.Id id);
```

Comment:
Should we make this and `MaxBucketsPerMeterLookup#getMaxBuckets` `@Nullable`?
In the current case the lookup should provide a fallback, if we make it nullable we can do that for the user.

Reply:
I think allowing nulls and falling back to the default value makes sense.

Reply:
I've updated the signature along with incorporating this change such that the mapping function should not deal with the default value other than to return `null`. This feels cleaner than the expectation of implementations calling specific methods on OtlpConfig